examples/voice-pipeline-agent/multi_task/restaurant_agent.py

```diff
@@ -0,0 +1,330 @@
+        """Called to start a new order."""
+        agent = AgentCallContext.get_current().agent
+        agent.user_data.clear()
+        logger.info("Started a new order")
+        return "Started a new order"
+
+
+"""
+Another way to create a task
+
+@llm.ai_callable()
+async def start_new_order() -> str:
+    ...
+
+def can_enter_greeter(agent: VoicePipelineAgent) -> bool:
+    return True
+
+greeter = AgentTask(
+    name="greeter",
+    instructions="...",
+    functions=[start_new_order],
+    options=AgentTaskOptions(
+        can_enter_cb=can_enter_greeter,
+        before_enter_cb=before_enter_cb,
+    ),
+)
+"""
+
+
+class OrderTaking(AgentTask):
+    def __init__(self, menu: str):
+        super().__init__(
+            name="order_taking",
+            instructions=(
+                "You are a professional order taker at a restaurant. Your tasks:\n"
+                f"1. Take orders from our menu: {menu}\n"
+                "2. Clarify special requests\n"
+                "3. Confirm order accuracy\n\n"
+                "Transfer to:\n"
+                "- customer_registration: when order is confirmed\n"
+                "- greeter: for general questions or starting over\n\n"
+                "Important:\n"
+                "- Use update_order function to save the order\n"
+                "- Ensure order is complete before transferring to customer details\n"
+                "- For non-order questions, transfer to greeter"
+            ),
+            functions=[self.update_order],
+            options=AgentTaskOptions(
+                can_enter_cb=self.can_enter,
+                before_enter_cb=before_enter_cb,
+            ),
+        )
+
+    def can_enter(self, agent: "VoicePipelineAgent") -> bool:
+        checked_out = agent.user_data.get("checked_out", False)
+        return not checked_out
+
+    @llm.ai_callable()
+    async def update_order(
+        self,
+        items: Annotated[
+            list[str],
+            llm.TypeInfo(description="The items of the full order"),
+        ],
+    ) -> str:
+        """Called when the user updates their order."""
+
+        agent = AgentCallContext.get_current().agent
+        agent.user_data["order"] = items
+
+        logger.info("Updated order", extra={"order": items})
+        return f"Updated order to {items}"
+
+
+class CustomerRegistration(AgentTask):
+    def __init__(self):
+        super().__init__(
+            name="customer_registration",
+            instructions=(
+                "You are collecting customer information for their order. Your tasks:\n"
+                "1. Get and confirm customer's name and comfirm the spelling\n"
+                "2. Get phone number and verify it's correct\n"
+                "3. Repeat both pieces of information back to ensure accuracy\n"
+                "Transfer to:\n"
+                "- checkout: when all details are confirmed\n"
+                "- order_taking: to modify the order\n"
```

Comment:
IMO embedding these in instructions is less than ideal.. it seems fragile (i.e. changing one place and not changing the other) since there are no typechecks/compilers

Reply:
I agree. maybe enhance the description of the transfer function instead of embedding this in separate instructions.

Reply:
Updated the detailed transfer instructions and removed them from the task instructions, but still need to add `Transfer to the next step using functions after ...` to each task to ensure automatic transfer between tasks.